File: BudgetTracker.API/Models/CategoryModel.cs
using System.ComponentModel.DataAnnotations;

namespace BudgetTracker.API.Models
{
    public class CategoryModel
    {
        [Key]
        public int CategoryID { get; set; }

        [Required(ErrorMessage = "O campo Título é obrigatório.")]
        public string ?Title { get; set; }

        public string ?Description { get; set; }

        [Range(0, double.MaxValue, ErrorMessage = "O valor deve ser maior que zero.")]
        public double TotalAmount { get; set; }
        public int Entries { get; set; }

        public int Icon { get; set; }
    }
}

File: budget_tracker/lib/models/category_model.dart
class CategoryModel {
  int? categoryID;
  String categoryName;
  String? categoryDescription;
  int? entries;
  double? totalAmount;
  int? icon;

  CategoryModel({
    this.categoryID,
    required this.categoryName,
    this.categoryDescription,
    this.entries,
    this.totalAmount,
    this.icon,
  });

  factory CategoryModel.fromJson(Map<String, dynamic> json) {
    return CategoryModel(
      categoryID: json['categoryID'] as int?,
      categoryName: json['title'] as String,
      categoryDescription: json['description'] as String?,
      entries: json['entries'] as int?,
      totalAmount: json['totalAmount'] != null
          ? (json['totalAmount'] as num).toDouble()
          : null,
      icon: json['icon'] as int?,
    );
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = {
      'title': categoryName,
    };

    if (categoryID != null) {
      data['categoryID'] = categoryID;
    }
    if (categoryDescription != null) {
      data['description'] = categoryDescription;
    }
    if (entries != null) {
      data['entries'] = entries;
    }
    if (totalAmount != null) {
      data['totalAmount'] = totalAmount;
    }
    if (icon != null) {
      data['icon'] = icon;
    }

    return data;
  }
}
